### DISCUSSION TITLE
Phazerville Suite v1.5.1

### DISCUSSION THREAD

**Discussion by djphazer** [djphazer/O_C-Phazerville#7] [Announcements]
Updates and bugfixes! By popular demand, **Piqued** is now included in all builds (and **Meta-Q** has been left out)

### Included Apps:
* Main builds: **Calibr8or, Hemisphere, Piqued, Quadraturia, Harrington 1200, Enigma, Captain MIDI, Neural Network, Darkest Timeline**
* "+stock" build: **Hemisphere, Quantermain, Piqued, Acid Curds, Sequins, Quadraturia, Harrington 1200**

### Other changes:
* New App: [Calibr8or](https://github.com/djphazer/O_C-BenisphereSuite/wiki/Calibr8or) - 4-channel Quantizer + Pitch CV fine-tuning utility, based on a design spec from Chris Meyer
* Fix: Quantizer transpose (affects all apps that use the internal Braids quantizer)
* Fix: DualTM trigger output stays high for the standard pulse length before decaying
* Fix: Quadraturia frequency display
* Fix: properly reverse calibration for _FLIP_180_ build
  - enables switching between flipped and non-flipped without recalibrating
* MIDI In/Out functionality has been updated for latest Teensy library spec (some MIDI stuff is probably broken in v1.5)
* Tweaks to UI Button behavior
  - Long-press fires before button is released
  - some actions now fire on button-down instead of button-release
* Various UI tweaks and fixes, including some new icons from Discord user ph.xyz

If there's a particular combination of Apps/features you would like that I haven't provided, please let me know! It's just a matter of switching build flags.

<hr /><em>This discussion was created from the release <a href='https://github.com/djphazer/O_C-BenisphereSuite/releases/tag/PSv1.5.1'>Phazerville Suite v1.5.1</a>.</em>

**Comment by nickstokes** [comment]
Awesome work. The Calibr8tor app is great!

**Comment by noodleboy91** [comment]
Awesome work, thank you!! Can we get a +stock flipped build please?

**Reply by djphazer** [reply]
For sure, I can do that!

**Reply by noodleboy91** [reply]
Sweet! Thank you :)

**Reply by djphazer** [reply]
I added a "+stock_flipped" build to the release page ;)

**Reply by vestedaf** [reply]
I'm currently running the +stock flipped v1.5.1 and noticed while using Acid Curds that it doesn't save my chord settings between power off/on sessions. I'm long holding the right encoder, the normal process, and it seems like it's saving but it does not stick. Is this a lack-of-memory issue, or something else?

Great work, as always!

**Reply by djphazer** [reply]
@vestedaf oh no, thanks for the heads up! It's possible that there's not enough memory to save state for all the apps in that build... I'm going to go through and figure out exactly how much storage each app uses to be sure.

**Reply by djphazer** [reply]
If my calculations are correct, looks like v1.5.1+stock uses 576 bytes of EEPROM out of 996 bytes allocated for AppData, so that shouldn't be the problem... Acid Curds only needs 25 bytes. Teensy 3.2 has 2048 bytes total.

Of course, I haven't actually reproduced it myself yet.

**Reply by vestedaf** [reply]
@djphazer interesting, thanks for the specs. I reloaded the firmware and I'm still having the same issue. I have not tested other full size apps. I really only use Acid Curds as a cv -> chord generator, usually just triads, so it's not been a big deal to change it to TRI on startup. I'm interested to hear if you can recreate the issue.

**Comment by danieldixon-thetin** [comment]
Amazing work here!
Since you mentioned having different combination of app... In a "+ stock" would it be possible to swap Acid Curds with Viznutcracker? 
And in doing so, would it perhaps be possible to squeeze in Low Rents? I know Hemisphere has its own version but the real app is very entertaining :)

**Reply by djphazer** [reply]
Ok, I'll try to pull those ones in soon! Seems like we'll need a "+stock1" and "+stock2" to fit more things in.

The goal is to eventually have a dynamic build system, where you can pick the individual apps and applets yourself, and a file is compiled on-the-fly. Platform IO makes this possible. Looks like there are some efforts happening towards this goal in other forks...

**Reply by danieldixon-thetin** [reply]
Yes, I imagine most of us would just love to cherry-pick the apps in our bespoke build! For instance I'd be happy to drop a few Hemisphere apps I already have well covered in my system, who knows with what real advantage in terms of space though...

**Comment by dvvii** [comment]
Amazing work, thank you so much! I’ve just installed this and it feels like xmas. Currently exploring dualTM + EuclidX. Also ebb & lfo is such a great surprise (and beautiful name). The new navigation functionality takes a bit to learn but it’s so good, for example to back and forth between positions without looping through the entire applet. I’m extremely happy to see the Random Walk applet included too. It is an obvious function that’s been missing for a while… I dream about a fully featured Source of Uncertainty applet and this ticks almost all the boxes.

mix and match would be… I can’t even! Thanks 🙏🏽

**Reply by nickstokes** [reply]
Not sure if you are aware, but there is a mode where the selection doesn't 'loop'. I prefer this, it means I can slam the selector to one side, then click a few times to get to what I want using muscle memory.

**Reply by dvvii** [reply]
I didn’t know you could change that behavior. Is that from this fork or an earlier one? How do you do it? I use muscle memory a lot too, with LFO applet especially, but the cursor/set method has its advantages.

**Reply by djphazer** [reply]
I'm glad you're enjoying it!

I actually locked the edit cursor mode in 1.5.1... modal w/ wraparound. I previously preferred it without wraparound, so you can "slam" it to one side as @nickstokes says... but I decided to lock it for this release and see what people think. The next release will have a new config screen where you can change it!

**Comment by ghost** [comment]
would it be possible to get CopierMaschine and Quantermain along with Hemispheres in a build? this would mean i never have to switch firmwares again. thank you!

**Reply by djphazer** [reply]
Ok, I'll pull in CopierMaschine soon as well. Quantermain is already in there.

FWIW, have you tried running two instances of the "A"SR applet?

**Reply by ghost** [reply]
thanks! i have tried the applet and it's cool, i just prefer copiermaschine in some ways.

**Comment by PedalEffected** [comment]
This is amazing work, this is basically every feature I liked about multiple other firmwares - thank you so much for making this!

Piqued is back!!!

**Comment by CorradoMorgana** [comment]
Awesome...testing the VOR version.
Darkest Timeline doesn't respond to clock input. ****Update...works fine now. Hit reset. Could be a dodgy cable.
Hemispheres works fine and I am a very happy chappie

**Reply by djphazer** [reply]
Yay, we need more happy chappies! Stay tuned, v1.6 is right around the corner 😉

**Reply by CorradoMorgana** [reply]
Exciteee!
